### PR TITLE
Update: Extract computation of metric column total to an overridable method

### PR DESCRIPTION
**Pull request by daniellevinson**
The new public method is `computeColumnTotal`.

### CODE REVIEW

File: packages/core/addon/components/navi-visualizations/table.js [hunk 1]
```diff
@@ -66,63 +66,96 @@ export default Component.extend({
...
+   * Returns sum of values, may be overriden with your own logic
+   *
+   * @param {Array} data
+   * @param {String} metricName
+   * @param {Object} totalRow
+   * @param {Object} column
+   * @param {String} type - `grandTotal` || `subtotal`
+   * @returns {*} sum
+   */
+  computeColumnTotal(data, metricName /*, totalRow, column, type*/) {
```
Comment (balajimanoharan) [inline packages/core/addon/components/navi-visualizations/table.js]: Why do we want this to be a public method?

File: packages/core/addon/components/navi-visualizations/table.js [hunk 2]
```diff
@@ -66,63 +66,96 @@ export default Component.extend({
...
+    //add totalRow indication to meta
+    set(
+      totalRow,
+      '__meta__',
+      Object.assign({}, get(totalRow, '__meta__'), { isTotalRow: true })
+    );
+
+    //if partial data add indication to meta
+    if (hasPartialData) {
+      set(totalRow, '__meta__.hasPartialData', true);
```
Comment (balajimanoharan) [inline packages/core/addon/components/navi-visualizations/table.js]: What are we doing with this metadata? Do we still want to display the totals even if we have partial data?

File: packages/core/addon/components/navi-visualizations/table.js [hunk 3]
```diff
@@ -165,32 +200,35 @@ export default Component.extend({
...
-    }
+      if (!get(this, 'options.showTotals.grandTotal')) {
+        return tableData;
+      }
 
-    return [...tableData, this._computeTotal(rawData, 'grandTotal')];
-  }),
+      return [...tableData, this._computeTotal(rawData, 'grandTotal')];
+    }
+  ),
```
Comment (balajimanoharan) [inline packages/core/addon/components/navi-visualizations/table.js]: Can we revert these changes?

Comment (mattgraham1995) [reply packages/core/addon/components/navi-visualizations/table.js]: I think prettier made these changes

File: packages/core/.eslintrc.js [hunk 4]
```diff
@@ -9,7 +9,6 @@ module.exports = {
     browser: true
   },
   rules: {
-    indent: [2, 2, { VariableDeclarator: { var: 2, let: 2, const: 3 } }],
```
[the remark this answers is not in the document]
Comment (daniellevinson) [reply packages/core/.eslintrc.js]: Prettier changes indentations when committing the code and makes tests fail on linting.
I ran `node_modules/.bin/eslint --print-config addon | node_modules/.bin/eslint-config-prettier-check` and got 
```
The following rules are unnecessary or might conflict with Prettier:

- indent
```

File: packages/core/addon/components/navi-visualizations/table.js [hunk 5]
```diff
@@ -66,63 +66,96 @@ export default Component.extend({
...
+   * @param {String} metricName
+   * @param {Object} totalRow
+   * @param {Object} column
+   * @param {String} type - `grandTotal` || `subtotal`
+   * @returns {*} sum
+   */
+  computeColumnTotal(data, metricName /*, totalRow, column, type*/) {
+    return data.reduce((sum, row) => {
+      let number = Number(row[metricName]);
+      if (!Number.isNaN(number)) {
```
[the remark this answers is not in the document]
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/table.js]: 👍

File: packages/core/addon/components/navi-visualizations/table.js (hunk 2 above)
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/table.js]: Currently we're not displaying totals if we have partial data, I haven't changed this behavior. Only moved it out of the reduce function, setting it once per row is enough.

File: packages/core/addon/components/navi-visualizations/table.js (hunk 3 above)
Comment (daniellevinson) [reply packages/core/addon/components/navi-visualizations/table.js]: yeah, prettier... haven't changed anything here
